From 0b46a65fe74d090a2e73aa7945c233df90645b16 Mon Sep 17 00:00:00 2001
From: Dirk Leber <dirk.leber@avionic-design.de>
Date: Wed, 2 Mar 2016 15:17:18 +0100
Subject: [PATCH] Make docs,examples and tests optional.

Add options to disable building documentation, examples and tests.
Modified to match version 2.6.0.

Signed-off-by: Soeren Grunewald <soeren.grunewald@avionic-design.de>
Signed-off-by: Dirk Leber <dirk.leber@avionic-design.de>
---
 Makefile.am  | 13 ++++++++++---
 configure.ac | 34 ++++++++++++++++++++++++++++++++++
 2 files changed, 44 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 952a7f9..d65b267 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -50,12 +50,19 @@ EXTRA_DIST = \
 SUBDIRS = \
 	lib \
 	src \
-	doc \
-	examples \
 	po \
-	tests \
 	tools
 
+if BUILD_DOCUMENTATION
+SUBDIRS += doc
+endif
+if BUILD_EXAMPLES
+SUBDIRS += examples
+endif
+if BUILD_TESTS
+SUBDIRS += tests
+endif
+
 # Create the ChangeLog, but only if we're inside a git working directory
 
 ChangeLog: $(srcdir)/tools/git2cl
diff --git a/configure.ac b/configure.ac
index 4c519dc..e0181a5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -112,6 +112,40 @@ AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 AC_CHECK_FUNCS([dup2 isascii memset pow regcomp setlocale strchr strtol])
 
+AC_ARG_ENABLE([documentation],
+	[AS_HELP_STRING([--disable-documentation],
+		[Do not generate documentation files])],[
+	 case "${enableval}" in
+		yes) build_docs=yes ;;
+		no) build_docs=no ;;
+		*) AC_MSG_ERROR([bad value ${enableval} for --disable-documentation]) ;;
+	 esac
+	],[build_docs=yes]
+)
+AM_CONDITIONAL([BUILD_DOCUMENTATION], [test "x$build_docs" = "xyes"])
+
+AC_ARG_ENABLE([examples],
+	[AS_HELP_STRING([--disable-examples], [Do not build examples])],[
+	 case "${enableval}" in
+		yes) build_examples=yes ;;
+		no) build_examples=no ;;
+		*) AC_MSG_ERROR([bad value ${enableval} for --disable-examples]) ;;
+	 esac
+	],[build_examples=yes]
+)
+AM_CONDITIONAL(BUILD_EXAMPLES, test "x$build_examples" = "xyes")
+
+AC_ARG_ENABLE([tests],
+	[AS_HELP_STRING([--disable-tests], [Do not build tests])],[
+	 case "${enableval}" in
+		yes) build_tests=yes ;;
+		no) build_tests=no ;;
+		*) AC_MSG_ERROR([bad value ${enableval} for --disable-tests]) ;;
+	 esac
+	],[build_tests=yes]
+)
+AM_CONDITIONAL(BUILD_TESTS, test "x$build_tests" = "xyes")
+
 AC_CONFIG_FILES(
 Makefile
 doc/Makefile
-- 
2.5.0

